src/main/java/com/amazon/ion/impl/bin/WriteBuffer.java

```diff
@@ -1519,6 +1521,34 @@ public int writeFixedIntOrUInt(final byte[] value) {
         return value.length;
     }
 
+    /**
+     * Writes a FlexSym with a symbol id.
+     */
+    public int writeFlexSym(int sid) {
+        if (sid != 0) {
+            return writeFlexInt(sid);
+        } else {
+            writeByte((byte) 0x01);
+            writeByte((byte) 0x70);
```

Comment:
What do you think about changing the spec so that this is 0x90 instead? That way it at least aligns with the upper nibble type (9 = symbol, whereas 7 = timestamp)

Reply:
I like that, and I suspect that was probably the intention, but then parts of the spec were modified without fully updating other parts of the spec.